internal/discovery/foreign-cluster-operator/auth.go

```diff
@@ -11,6 +11,7 @@ import (
 	"github.com/liqotech/liqo/pkg/auth"
 	"github.com/liqotech/liqo/pkg/crdClient"
 	"github.com/liqotech/liqo/pkg/discovery"
+	discoveryPkg "github.com/liqotech/liqo/pkg/discovery"
```

Comment:
I think `somethingPkg` is pretty strange as an import name (it is obviously a package that you are importing).

Reply:
I removed where possible, there is a name collision in some packages

Reply:
do these collisions derive from importing both `internal/discovery` and `pkg/discovery` in the same files?

Reply:
yes, and because in the discovery internals the object that owns the functions has also "discovery" name